packages/actions/src/Concerns/CanImportRecords.php

```diff
@@ -409,7 +409,10 @@ public function getUploadedFileStream(TemporaryUploadedFile $file)
 
     protected function detectCsvEncoding(TemporaryUploadedFile $file): ?string
     {
-        $fileContents = file_get_contents($file->getRealPath());
+        $filePath = $file->getRealPath();
+        $stream = fopen($filePath, mode: 'r');
```

Comment:
Can we replace this with `getUploadedFileStream()` so that we can also support S3? Because I think it breaks S3 right now, maybe you could test that if you have time?

Reply:
I've removed the file opening operation from `getCsvEncoding()` to instead use the resource that's already being opened in `getUploadedFileStream()`, according to the user's storage driver. The `getCsvEncoding()` method now accepts the resource as a variable, and only *reads* the content. Unless I missed anything, it now works locally and for s3.